plugins/reports/articles/ArticleReportPlugin.inc.php

```diff
@@ -60,232 +55,221 @@ function display($args, $request) {
 
+		// Prepare for UTF8-encoded CSV output.
 		header('content-type: text/comma-separated-values');
 		header('content-disposition: attachment; filename=articles-' . $acronym . '-' . date('Ymd') . '.csv');
+		$fp = fopen('php://output', 'wt');
+		// Add BOM (byte order mark) to fix UTF-8 in Excel
+		fprintf($fp, chr(0xEF).chr(0xBB).chr(0xBF));
 
-		$articleReportDao = DAORegistry::getDAO('ArticleReportDAO'); /* @var $articleReportDao ArticleReportDAO */
-		list($articlesIterator, $authorsIterator, $editorsIterator, $decisionsIterator) = $articleReportDao->getArticleReport($journal->getId());
+		$submissionDao = DAORegistry::getDAO('SubmissionDAO'); /* @var $submissionDao SubmissionDAO */
+		$editDecisionDao = DAORegistry::getDAO('EditDecisionDAO'); /* @var $editDecisionDao EditDecisionDAO */
+		$stageAssignmentDao = DAORegistry::getDAO('StageAssignmentDAO'); /* @var $stageAssignmentDao StageAssignmentDAO */
+		$userGroupDao = DAORegistry::getDAO('UserGroupDAO'); /* @var $userGroupDao UserGroupDAO */
+		$userDao = DAORegistry::getDAO('UserDAO'); /* @var $userDao UserDAO */
+		$sectionDao = DAORegistry::getDAO('SectionDAO'); /* @var $sectionDao SectionDAO */
+		$submissionKeywordDao = DAORegistry::getDAO('SubmissionKeywordDAO'); /* @var $submissionKeywordDao SubmissionKeywordDAO */
+		$submissionSubjectDao = DAORegistry::getDAO('SubmissionSubjectDAO'); /* @var $submissionSubjectDao SubmissionSubjectDAO */
+		$submissionDisciplineDao = DAORegistry::getDAO('SubmissionDisciplineDAO'); /* @var $submissionDisciplineDao SubmissionDisciplineDAO */
+		$submissionAgencyDao = DAORegistry::getDAO('SubmissionAgencyDAO'); /* @var $submissionAgencyDao SubmissionAgencyDAO */
 
-		$maxAuthors = $this->getMaxCount($authorsIterator);
-		$maxEditors = $this->getMaxCount($editorsIterator);
-		$maxDecisions = 0;
-		foreach ($decisionsIterator as $decisionsIteratorForArticle) {
-			$maxDecisionsForArticle = $this->getMaxCount($decisionsIteratorForArticle);
-			if ($maxDecisionsForArticle > $maxDecisions) $maxDecisions = $maxDecisionsForArticle;
-		}
+		$editorUserGroupIds = array_map(function($userGroup) {
+			return $userGroup->getId();
+		}, array_filter($userGroupDao->getByContextId($journal->getId())->toArray(), function($userGroup) {
+			return in_array($userGroup->getRoleId(), [ROLE_ID_MANAGER, ROLE_ID_SUB_EDITOR]);
+		}));
 
 		AppLocale::requireComponents(LOCALE_COMPONENT_APP_EDITOR, LOCALE_COMPONENT_PKP_SUBMISSION, LOCALE_COMPONENT_PKP_READER);
 
-		import('classes.submission.Submission');
-
-		$columns = array(
-			'submission_id' => __('article.submissionId'),
-			'title' => __('article.title'),
-			'abstract' => __('article.abstract')
-		);
+		// Load the data from the database and store it in an array.
+		// (This must be stored before display because we won't know the data
+		// dimensions until it has all been loaded.)
+		$results = $sectionTitles = [];
+		$submissions = $submissionDao->getByContextId($journal->getId());
```

Comment:
It doesn't make a difference, really, but this could use the service class as `$submissionIterator = Services::get('submission')->getMany(['contextId' => $journal->getId()]);`